pkg/processor/test/suite/suite.go

```diff
@@ -82,19 +87,32 @@ type BlastConfiguration struct {
 // SetupSuite is called for suite setup
 func (suite *TestSuite) SetupSuite() {
 	var err error
+
 	if suite.RuntimeDir == "" {
 		suite.RuntimeDir = suite.Runtime
 	}
 
-	common.SetVersionFromEnv()
+	if suite.PlatformType == "" {
+		suite.PlatformType = "local"
+	}
+
+	if suite.Namespace == "" {
+		suite.Namespace = "default"
+	}
+
+	// this will preserve the current behavior where function names are renamed to be unique upon deployment
+	suite.FunctionNameUniquify = true
 
 	suite.Logger, err = nucliozap.NewNuclioZapTest("test")
 	suite.Require().NoError(err)
 
 	suite.DockerClient, err = dockerclient.NewShellClient(suite.Logger, nil)
 	suite.Require().NoError(err)
 
-	suite.Platform, err = factory.CreatePlatform(suite.Logger, "local", nil, "default")
+	suite.Platform, err = factory.CreatePlatform(suite.Logger,
+		suite.PlatformType,
+		suite.PlatformConfiguration,
+		suite.Namespace)
```

Comment:
I think the nuclio tests are a bit bloated as they are with knobs and fields
Can you please decouple the tests as much as possible, it's best to just write those setup lines again instead of constraining test suites structure and signature to each other. 
The "duplicated" code is very short and basic, IMO each suite can do this, and instead the suites are now bloated with fields which are only needed once.
No need for the test suite to externalize so many things (like platform type and namespace) when a single suite can declare this explicitly and be readable, a lot of things are used once here, in the setup
These are needed only if the suite share code in a way that requires this (and this can also be resolved with calls to library functions)
This can be done in a separate PR